tests/active_features/connection-set.feature

```diff
@@ -0,0 +1,37 @@
+Feature: Set a site's connection mode
+  In order to ensure the correct sort of connectivity for my site
+  As a user
+  I need to be able to change my site's connection mode.
+
+  Background: I am authenticated and I have a site named [[test_site_name]]
+    Given I am authenticated
+    And a site named "[[test_site_name]]"
+
+  @vcr site_set-connection-mode_git
+  Scenario: Setting connection mode to git
+    When I run "terminus connection:set [[test_site_name]].dev git"
+    Then I should see a notice message: Enabling git push mode for "dev"
+
+  @vcr site_set-connection-mode_sftp
+  Scenario: Setting connection mode to sftp
+    When I run "terminus connection:set [[test_site_name]].dev sftp"
+    Then I should see a notice message: Enabling on-server development via SFTP for "dev"
+
+  @vcr site_set-connection-mode_git
+  Scenario: Failing to set the connection mode to the current sftp mode
+    # Note: The VCR fixture has the environment in sftp mode to start. Want a given like:
+    # Given the [[test_site_name]].dev environment is in the sftp connection mode
+    When I run "terminus connection:set [[test_site_name]].dev sftp"
+    Then I should see a notice message: The connection mode is already set to sftp.
+
+  @vcr site_set-connection-mode_sftp
+  Scenario: Failing to set the connection mode to the current git mode
+    # Note: The VCR fixture has the environment in git mode to start. Want a given like:
+    # Given the [[test_site_name]].dev environment is in the git connection mode
+    When I run "terminus connection:set [[test_site_name]].dev git"
+    Then I should see a notice message: The connection mode is already set to git.
+
+  Scenario: Attempting to set connection mode to an invalid mode
+    When This step is implemented I will test: setting invalid connection modes
```

Comment:
Note: I was looking for a way to mark a Scenario as a pending test so that we could define some ACs that we know we want but aren't able to instrument at the moment for whatever reason. I was hoping for a standard annotation such as `@pending` but it doesn't look like that is a thing just yet in behat.

This is what I came up with and it skips all the steps **that follow** by throwing a `Behat\PendingException` and yields the string following the colon, in this case the message: `Testing setting invalid connection modes is pending.`

I'm not married to the implementation or the specific language in the step match pattern but I do really like that we can write pending ACs that are pesky (not just TODO comments) that don't break the build.